pkg/controller/garbagecollector/garbagecollector_test.go

```diff
@@ -920,6 +921,107 @@ func TestGarbageCollectorSync(t *testing.T) {
 	if err != nil {
 		t.Fatalf("Expected garbagecollector.Sync to still be running but it is blocked: %v", err)
 	}
+
```

Comment:
Above this (github won't let me comment), there is a line: `// The pseudo-code of GarbageCollector.Sync():` - this needs updating to no longer have `wait.PollImmediateUntil() loops with 100ms (hardcode) util the `stopCh` is closed:` (which is removed in this PR)